src/client.js

```diff
@@ -47,7 +49,10 @@ if (module.hot) {
     store = new NewIndexStore();
     store.initialize(data);
 
+    ReactDOM.unmountComponentAtNode(document.getElementById('app'));
+    store.cache.setPerma(true);
     renderAll();
+    store.cache.setPerma(false);
```

Comment:
why do we renderAll with the store setPerma to true? What happens if the page is in the middle of a getDependencies call to render the new page?